api/v1/submissions/PKPSubmissionHandler.inc.php

```diff
@@ -137,6 +153,16 @@ public function __construct()
                     'handler' => [$this, 'versionPublication'],
                     'roles' => [Role::ROLE_ID_MANAGER, Role::ROLE_ID_SUB_EDITOR, Role::ROLE_ID_ASSISTANT],
                 ],
+                [
+                    'pattern' => $this->getEndpointPattern() . '/{submissionId:\d+}/publications/{publicationId:\d+}/contributors',
+                    'handler' => [$this, 'addContributor'],
+                    'roles' => [Role::ROLE_ID_MANAGER, Role::ROLE_ID_SUB_EDITOR, Role::ROLE_ID_ASSISTANT],
+                ],
+                [
+                    'pattern' => $this->getEndpointPattern() . '/{submissionId:\d+}/publications/{publicationId:\d+}/contributors/saveOrder',
+                    'handler' => [$this, 'saveContributorsOrder'],
+                    'roles' => [Role::ROLE_ID_MANAGER, Role::ROLE_ID_SUB_EDITOR, Role::ROLE_ID_ASSISTANT],
+                ],
```

Comment:
You'll need to check, but I think that an author who has permission to edit metadata can edit contributors. This is a setting in the "participant asssignment" form in the workflow. If that's the case, you'll need to add `Role::ROLE_ID_AUTHOR` to all `PUT`, `POST` and `DELETE` methods.

Reply:
This needs to be considered and tested it as well.

Reply:
Changed. Will be tested